packages/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { Button, Typography, WithStyles, withStyles } from "@material-ui/core";
+import { ButtonProps } from "@material-ui/core/Button";
+import clsx from "clsx";
+import * as React from "react";
+
+import { FilterBarActiveFilterBadge, FilterBarActiveFilterBadgeProps } from "../filterBarActiveFilterBadge/FilterBarActiveFilterBadge";
+import { FilterBarButtonClassKey, styles } from "./FilterBarButton.styles";
+
+export interface FilterBarButtonProps extends ButtonProps {
+    dirtyFieldsBadge?: React.ComponentType<FilterBarActiveFilterBadgeProps>;
+    countValue?: number;
+    openPopover: boolean;
+}
+
+const FilterBarButton = ({
+    children,
+    dirtyFieldsBadge,
+    countValue,
+    openPopover,
+    classes,
+    ...buttonProps
+}: FilterBarButtonProps & WithStyles<typeof styles>): React.ReactElement => {
+    const showFilterBadge = !!(countValue && countValue > 0);
+    const FilterBarActiveFilterBadgeComponent = dirtyFieldsBadge ? dirtyFieldsBadge : FilterBarActiveFilterBadge;
+
+    return (
+        <Button
+            className={clsx(classes.root, showFilterBadge && classes.selected && classes.withFilterBadge, openPopover && classes.open)}
```

Comment:
Isn't `classes.selected` redundant needed here?  

It looks to me like: `if (showFilterBadge  && classes.selected) { className += classes.withFilterBadge }` 🤔